src/map/path.cpp

```diff
@@ -431,46 +461,22 @@ bool path_search(struct walkpath_data *wpd, int16 m, int16 x0, int16 y0, int16 x
 	return false;
 }
 
-
-//Distance functions, taken from http://www.flipcode.com/articles/article_fastdistance.shtml
 bool check_distance(int dx, int dy, int distance)
 {
 #ifdef CIRCULAR_AREA
 	//In this case, we just do a square comparison. Add 1 tile grace for diagonal range checks.
-	return (dx*dx + dy*dy <= distance*distance + (dx&&dy?1:0));
+	return (euclidean_range(dx,dy) <= std::pow(distance, 2) + (dx && dy ? 1 : 0));
 #else
-	if (dx < 0) dx = -dx;
-	if (dy < 0) dy = -dy;
-	return ((dx<dy?dy:dx) <= distance);
+	return (chebyshev_range(dx, dy) <= distance);
 #endif
 }
 
 unsigned int distance(int dx, int dy)
 {
 #ifdef CIRCULAR_AREA
-	unsigned int min, max;
-
-	if ( dx < 0 ) dx = -dx;
-	if ( dy < 0 ) dy = -dy;
-	//There appears to be something wrong with the approximation below when either dx/dy is 0! [Skotlex]
-	if ( dx == 0 ) return dy;
-	if ( dy == 0 ) return dx;
-
-	if ( dx < dy )
-	{
-		min = dx;
-		max = dy;
-	} else {
-		min = dy;
-		max = dx;
-	}
-   // coefficients equivalent to ( 123/128 * max ) and ( 51/128 * min )
-	return ((( max << 8 ) + ( max << 3 ) - ( max << 4 ) - ( max << 1 ) +
-		( min << 7 ) - ( min << 5 ) + ( min << 3 ) - ( min << 1 )) >> 8 );
+	return std::max(chebyshev_distance(dx, dy), (unsigned)1);
```

Comment:
if anyone give-me okay i'll change it to euclidean_distance because this are on #ifdef circular_area, idk why it using chebyshev bit mode